File: Models/Topic.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Models
{
    public partial class Topic
    {
        public Topic()
        {
            Replies = new HashSet<Reply>();
        }

        public int TopicId { get; set; }
        public string TopicName { get; set; }
        public string TopicBody { get; set; }
        public int? ProfileId { get; set; }
        public DateTime? PostTimestamp { get; set; }
        public int? CategoryId { get; set; }

        public virtual Category Category { get; set; }
        public virtual Profile Profile { get; set; }
        public virtual ICollection<Reply> Replies { get; set; }
    }
}

File: pawpers-angular/src/app/topic-page/topic.ts
export interface Topic {
  topicId: number
  topicName: string
  topicBody: string
  profileId: number
  postTimestamp: any
  categoryId: number
  category: Category
  profile: Profile
  replies: Reply[]
}

export interface Category {
  categoryId: number
  categoryName: string
  topics: Topic[]
}

export interface Profile {
  profileId: number
  profileName: string
  profileStreetaddress: string
  profileCity: string
  profileStateid: number
  profileZipcode: string
  profileAge: number
  profileHomephone: string
  profilePersonalphone: string
  profileEmail: string
  profileOccupation: string
  profileSpousename: string
  profileChildren: number
  profileDwellingid: number
  profileHasyard: number
  profileLandlordname: string
  profileLandlordphone: string
  profileOtherpetname: string
  profileOtherpetbreed: string
  profileOtherpetsex: string
  profileOtherpetage: number
  profileFamilyallergies: number
  profileResponsiblefordog: string
  profileAdoptionreason: string
  profileDogsleepat: string
  profileDogaggresive: string
  profileMedfordog: string
  profileNocaredog: string
  profileDwelling: any
  profileState: any
  favorites: Favorite[]
  replies: Reply[]
  topics: Topic[]
}

export interface Reply {
  replyId: number
  replyMessage: string
  replyTimestamp: any
  profileId: number
  topicId: number
  profile: Profile
  topic: Topic
}

export interface Favorite {
  favId:       number;
  dogId:       number;
  isAvailable: number;
  profileId:   number;
  profile:     Profile;
}
